salt/modules/opkg.py

```diff
@@ -582,6 +591,12 @@ def install(name=None,
             info={'errors': errors, 'changes': ret}
         )
 
+    if list_pkgs_errors:
+        raise CommandExecutionError(
+            'Problem encountered after installing package(s). Cannot provide changes list.',
+            info={'errors': list_pkgs_errors}
+        )
+
     _process_restartcheck_result(rs_result, **kwargs)
```

Comment:
I think we should let this function be executed before checking for list_pkgs_errors. If the packages were installed with success and they need a reboot we should mark that even if opkg list-installed failed, otherwise the user will see that the packages are installed but not working (because a reboot was required but was not flagged). What do you think?

Reply:
I agree